sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java

```diff
@@ -270,6 +272,8 @@ public void populateDisplayData(DisplayData.Builder builder) {
     private static final Logger LOG = LoggerFactory.getLogger(BoundedSourceAsSDFWrapperFn.class);
     private static final long DEFAULT_DESIRED_BUNDLE_SIZE_BYTES = 64 * (1 << 20);
 
+    private final ArrayDeque<Closeable> restrictionTrackerStack = new ArrayDeque<>(5);
```

Comment:
Not obvious to me why use a Dequeue of size 5 here.

Reply:
To be honest, the number was based on a very vague theory, and I had hoped to learn more about it in this PR.